test/gtest/conv_immediate.cpp

```diff
@@ -0,0 +1,532 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest_common.hpp>
+#include <miopen/env.hpp>
+#include <miopen/handle.hpp>
+
+#include <cstdlib>
+#include <ctime>
+#include <functional>
+#include <numeric>
+#include <type_traits>
+#include <vector>
+
+#include <miopen/convolution.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_layout.hpp>
+#include <miopen/bfloat16.hpp>
+#include <half/half.hpp>
+#include "driver.hpp"
+#include "tensor_holder.hpp"
+#include "cpu_conv.hpp"
+#include "cartesian_product.hpp"
+
+using float16 = half_float::half;
+
+namespace {
+
+static constexpr int RAND_INTEGER_MAX       = 5;
+static constexpr int RAND_INTEGER_MIN       = -4;
+static constexpr float MAX_INTEGER_INTERVAL = 4.f;
+
+template <typename T>
+auto gen_value =
+    [](auto... is) { return static_cast<T>(prng::gen_A_to_B(RAND_INTEGER_MIN, RAND_INTEGER_MAX)); };
```

Comment:
Mostly just curious, but is there a reason to use a templated lambda variable instead of a method for the generator?

Reply:
If you pass is into `.generate()` method, it supplies input lambda with indexes based on the tensor's dimensions. `prng::` generator does not take any arguments, so we have to wrap it into lambda which ignores input indexes and just returns random value for each call.

Reply:
Oh sorry, I meant I just found it surprising that this was a lambda variable instead of a method like:

```cpp
template <typename T>
auto GenValue(auto... is){ return static_cast<T>(prng::gen_A_to_B(RAND_INTEGER_MIN, RAND_INTEGER_MAX)); }
```

Reply:
This was existing code, and I don't have an opinion. I went with the conservation of effort approach and simply relocated it.